File: fixtures/sample.go
// Penumbra fixture — Go (tint: cyan)
package main

import (
	"errors"
	"fmt"
)

type User struct {
	ID     int
	Name   string
	Active bool
	Tags   []string
}

var ErrInactive = errors.New("inactive")

func (u *User) Greet() (string, error) {
	if !u.Active {
		return "", ErrInactive
	}
	return fmt.Sprintf("Hello, %s (%d tags)", u.Name, len(u.Tags)), nil
}

func main() {
	users := []*User{
		{ID: 1, Name: "Ada", Active: true, Tags: []string{"admin"}},
		{ID: 2, Name: "Grace", Active: false},
	}
	for _, u := range users {
		if msg, err := u.Greet(); err != nil {
			fmt.Printf("<%s: %v>\n", u.Name, err)
		} else {
			fmt.Println(msg)
		}
	}
}

File: fixtures/sample.py
# Penumbra fixture — Python (tint: amber)
from dataclasses import dataclass, field
from typing import Optional, List


@dataclass
class User:
    id: int
    name: str
    active: bool = True
    tags: List[str] = field(default_factory=list)


def greet(user: User) -> Optional[str]:
    """Return a greeting if the user is active, else None."""
    if not user.active:
        return None
    return f"Hello, {user.name} ({len(user.tags)} tags)"


users = [
    User(1, "Ada", tags=["admin"]),
    User(2, "Grace", active=False),
]

if __name__ == "__main__":
    for u in users:
        msg = greet(u)
        print(msg or f"<inactive: {u.name}>")
